casper/src/test/resources/RevAddressTest.rho

```diff
@@ -62,5 +63,13 @@ in {
       revAddress!("fromPublicKey", "F0F0".hexToBytes(), *retCh) |
       rhoSpec!("assert", (Nil, "== <-", *retCh), "correct RevAddress", *ackCh)
     }
+  } |
+
+  contract test_fromUnfName(rhoSpec, _, ackCh) = {
+    new unf1, unf2, r1, r2 in {
+      revAddress!("fromUnfName", *unf1, *r1) |
+      revAddress!("fromUnfName", *unf2, *r2) |
+      rhoSpec!("assert", (*r1, "!= <-", *r2), "expecting RevAddresses to be unequal", *ackCh)
```

Comment:
we need a negative case too: pass a forgeable process (e.g. 42) and expect the result to be Nil

Reply:
Can do but this would be inconsistent with how most (all?) system processes work. For example if you pass an unforgeable name to `RevAddress!fromPublicKey` it won't return `Nil` but will just stay in the tuple space.

Reply:
I though I've seen a test saying otherwise:
https://github.com/rchain/rchain/blob/8bd29248267b484ad79e410207f7fe0c6694f89a/casper/src/test/resources/RevAddressTest.rho#L64

Reply:
That is because `RevAddress!fromPublicKey` takes Rholang ByteArrays. Iff the ByteArray passed to `RevAddress!fromPublicKey` is not a valid PublicKey, then we return `Nil`. If you pass a Rholang String to `RevAddress!fromPublicKey` you won't get anything back though.

Reply:
In both cases, not returning anything is the worst thing we can do. We should return Nil in both of those cases, if not an either with an error message.